proxy.go

```diff
@@ -40,14 +60,16 @@ func ProxyHandler(w http.ResponseWriter, req *http.Request) {
 	// by url.Parse and the server handler
 	plain, _ := http.NewRequest(req.Method, req.URL.String(), req.Body)
 	// create a new client
-	client := NewClient(insecureSkipVerify)
-	r, err := client.Do(plain)
+
+	r, err := p.HttpClient.Do(plain)
 	if err != nil {
 		Logger.Println("Request execution error:", err)
 		w.WriteHeader(500)
 		w.Write([]byte(err.Error()))
 		return
 	}
+	// Close the connection to reuse it
+	defer r.Body.Close()
```

Comment:
Noting the Go docs for the response body, I think this actually has to be closed before the request is retried on L95.